readme.txt
=== Mashshare Social Sharing Optimization ===
Author URL: https://www.mashshare.net
Plugin URL: https://www.mashshare.net
Contributors: ReneHermi
Donate link: https://www.mashshare.net/buy-me-a-coffee/
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html
Tags: Mashable, , Social Sharing, Social sharing optimization, Share button, share buttons, Facebook Share button, Twitter Share Button, Social Share, Social buttons, Share, Share this, Google+, Twitter, Facebook, Digg, Stumble Upon, Linkedin, +1, add to any, AddThis, add this, addtoany, admin, aim, analytics, bookmark, bookmarking, bookmarks, bookmark button, bookmark menu, buddypress, chrome extension, comments, del.icio.us, Diigo, buffer, button, del.icio.us, e-mail, email, email button, Facebook, facebook like, facebook recommend, facebook share, facebook share button, fark, firefox extension, flickr, floating buttons, follow, foursquare, gmail, goo.gl, google, google analytics, google plus, google plus one, google widget, hackernews, Hubspot, hyves, instagram, instapaper, NewsVine, odnoklassniki, pocket, popular, outbrain, PrintFriendly, popular posts, print, Whatsapp, rss, related posts, related content, share links, share bar, sharebar, social analytics, velocity graph, social networking, share and enjoy, sidebar, Vkontakte, technorati, socialize, url shortener, sonico, statistics tumblr, vimeo, yarrp, youtube, recommendations, posterous, orkut, icon, icons, image, images, Like, links, lockerz, page, pages, pin, pin it, pinit, pinterest, plugin, plus 1, plus one, Post, posts, Reddit, save, seo, Share, Shareaholic, sharedaddy, sharethis, sharing, shortcode, sidebar, sociable, social, social bookmarking, social bookmarks, statistics, stats, stumbleupon, svg, technorati, tumblr, tweet, twitter, vector, widget, ,WPSSO, wpmu
Requires at least: 3.1+
Tested up to: 4.0
Stable tag: 2.1.2

Mashshare social sharing buttons, a high-performance social sharing optimization ecosystem for Facebook Twitter Subscribe and 17 more networks 

== Description == 

> Mashshare Share Buttons shows the total share counts of Facebook and Twitter and a big subscribe button at a glance per default.
It puts some beautiful and clean designed Share Buttons on top and end of your posts to get the best most possible social share feedback from your user.
It´s inspired by the Share buttons Mashable.com is using.

Free and premium Add-Ons available for Google Analytics, more Social Networks, responsive design, Velocity Graph, Sticky Share bar, Social Sharing Optimization and much more.

<h3> Mashshare demo </h3>

[Demo link Mashshare Share Buttons](https://www.mashshare.net/?ref=1 "Share-Buttons - Mashable inspired Share Buttons")

This plugin is in active development and will be updated on a regular basis - Please do not rate negative before i tried my best to solve your issue.


= Main Features =

* High Performance - easy to use - Share Buttons for the most common networks
* High Resolution lossless vector font icons
* Show the Total Share count at a glance
* Object and transient caches to provide incredibly fast execution speed
* Shortcodes
* Extensible with many [Add-Ons](https://www.mashshare.net/downloads) (Google Analytics, More Social Networks, Responsive, Velocity Graph and more...)
* Developer friendly with several filters and actions.

= All Features: =

<strong>New Version 2.x</strong>

* Improved performance
* Option to disable share count completely  (no sql queries will be generated any longer)
* Shortcode option to disable share counts
* Check if curl is working on the server
* Option to disable share cache for testing purposes
* Use of sharp and crisp clear font icons instead png icons
* Button 'extra content' for content slider subcribe forms or any other content New: Use a link behind the Subscribe button instead the toggle slider
* Complete rewrite of css for easier modifications
* Improved extension system
* Improved backend, new Add-On page
* Multi language capable, *.po files
* Change color of share counts via setting
* Count up animation for share counts (Does not work for shortcodes and on blog pages)
* HTML5 Tag <aside> wrapped around to tell search engines that the share buttons are not part of the content
* Plus button moves to end of share buttons when activated and does not stay longer in place.
* Drag and drop sort order of services.
* Enable desired services with one click
* Choose which network should be visible all the time This ones will be large sized by default. Other ones are behind the plus sign
* Two different share button styles includes
* Choose border radius of the buttons from settings
* Keep settings when plugin is uninstalled - optional
* Custom CSS field for individual styling of the share buttons

<strong> Add-Ons available for </strong>

* Google / G+
* Whatsapp (Whatsapp button only shown on iOS devices)
* Pinterest
* Digg
* Linkedin
* Reddit
* Stumbleupon
* Vk / VKontakte
* Print
* Delicious
* Buffer
* Weibo
* Pocket
* Xing
* Tumblr
* Mail

= High Performance =

Mashshare is *coded well and developed for high performance*. It´s making full use of available persistant and non persistant caching techniques.
Mashshare loads only the Javascript and php object classes it needs at the moment of execution, making it small and fast and easy extensible by third party developer.

<blockquote>
<h4>How proven fast is Mashshare?</h4>

<p>This are a few example metrics from the <a href="https://wordpress.org/plugins/p3-profiler/">P3 (Plugin Performance Profiler)</a> plugin, using <a href="http://wptest.io/">WP Test Data</a>, and the default settings of several popular plugins, 
along with the WordPress Core and Twenty Fourteen theme for reference purposes:

Thanks to [JS Morisset](https://profiles.wordpress.org/jsmoriss/ "Social Sharing benchmark") for the idea of creating a common list of benchmarked plugins</p>

<ul>
        <li><strong>0.0166</strong> secs - <em><strong>Mashshare Share buttons v2.1.2 (no Add-Ons)</strong></em></li>
	<li><strong>0.0174</strong> secs - Facebook v1.5.5</li>
	<li><strong>0.0176</strong> secs - WPSSO Social Sharing Buttons (SSB) v1.0.10 + WPSSO</li>
	<li><strong>0.0181</strong> secs - MailPoet Newsletters (aka Wysija Newsletters) v2.6.11</li>
	<li><strong>0.0258</strong> secs - All in One SEO Pack v2.2.3.1</li>
	<li><strong>0.0293</strong> secs - JetPack by WordPress.com v2.9.2 ()</li>
	<li><strong>0.0342</strong> secs - WP e-Commerce v3.8.13.3 (No Products)</li>
	<li><strong>0.0348</strong> secs - WordPress Social Sharing Optimization (WPSSO) v2.6.8</li>
	<li><strong>0.0416</strong> secs - WordPress SEO by Yoast v1.6.3 </li>
	<li><strong>0.0457</strong> secs - NextGEN Facebook (NGFB) v7.6.1.1</li>
	<li><strong>0.0730</strong> secs - MarketPress Lite - WordPress eCommerce v2.9.5.3 (No Products)</li>
        <li><strong>0.0749</strong> secs - bbPress v2.5.4 (No Forums or Topics)</li>
	<li><strong>0.0800</strong> secs - Easy Digital Downloads v2.1.5 (No Products - no Add-Ons)</li>
	<li><strong>0.0924</strong> secs - <strong>WordPress Twenty Fourteen Theme v1.1</strong></li>
	<li><strong>0.1003</strong> secs - SEO Ultimate v7.6.5.1</li>
	<li><strong>0.1021</strong> secs - WooCommerce v2.2.6 (No Products)</li>
	<li><strong>0.1092</strong> secs - <strong>WordPress Core v4.0.0</strong></li>
	<li><strong>0.1404</strong> secs - BuddyPress v2.1.1 (No Activities)</li>
	<li><strong>0.1404</strong> secs - NextGEN Gallery by Photocrati v2.0.66.29 (No Galleries or Images)</li>
</ul>

<p><small><em>This Benchmarks were executed on Intel i5 2.3Ghz, 8GB DDR3, using an APC opcode/object cache without any DB or file Cache Plugin like W3Total, WordPress v4.0, and P3 v1.5.2 ( Attempt to disable opcode optimizers - Otherwise some plugins can not be tested) and in manual mode to make sure that
for every plugin the same page is loaded. Automatic excecution of P3 would randomly load WordPress pages which is inaccurate for testing on same conditions. All plugins were benchmarked on sepearate instances with no other plugins enabled.
The benchmarks were runned a few times to make sure that any internal plugin caching function is fired. E.g. Mashshare gets only really fast after second load of a page because the sharecount must be returned from external service first and than stored into internal object cache.
This benchmarks give only a vast overview of performance differences and results will vary on other WordPress installations depending on different hardware and software configurations.
</em></small></p>
<strong>Please note:</strong>
For this test the P3 Option 'Attempt to disable opcode optimizers' is enabled (Otherwise some plugin benchmarks returns no benchmark times ). 
So real execution time of every single Plugin is faster (probaby with a similar factor for all benchmarked plugins) but can not be measured exactely with my testing environment.
<p></p>
</blockquote>

**Shortcodes**

* Use `[mashshare]` anywhere in pages or post's text to show the buttons and total count where you like to at a custom position.
Buttons are shown exactly on the place where you use the shortcode in your content.

There are more parameters available:

 `[mashshare shares="false" buttons="true" align="left"]` for buttons without shares, alignment left
 `[mashshare shares="true" buttons="false" align="right"]` for sharecount without buttons, alignment right

* For manual insertion of the Share Buttons in your template files use the following php code where you want to show your Mash share buttons:`do_action('mashshare');`
Configure the Share buttons sharing function in the settings page of the plugin.
* Change the color of Mashshare count with setting option.

**Full SEO third party plugin support**
Mashshare integrates with [All in One SEO Pack](http://wordpress.org/plugins/all-in-one-seo-pack/) and [WordPress SEO by Yoast](http://wordpress.org/plugins/wordpress-seo/).


= How does it work? =

Mashshare makes use of the great webservice sharedcount.com and periodically checks for the total count 
of all your Facebook and Twitter shares and cumulates them. It than shows the total number beside the Share buttons. 
No need to embed dozens of external slow loading scripts into your website. 
 
= How to install and setup? =
Install it via the admin dashboard and to 'Plugins', click 'Add New' and search the plugins for 'Mashshare'. Install the plugin with 'Install Now'.
After installation goto the settings page Settings->Mashshare and make your changes there.


== Frequently Asked Questions ==

> Find here the Frequently Asked Questions. If your question is not answered look at:
https://www.mashshare.net/faq/

<h4>There are no social share buttons visible after updating or installing Mashshare</h4>
This happens ins most cases when you are using the Mashshare Network Add-On which is disabled during update process or when your are updating from a very early Mashshare version 1.x.
Solution: Disable Mashshare Network Add-On and Mashshare. Enable Mashshare THAN the Network Add-On and all Share buttons become visible again. (Activating order is important here)

<h4>Why is the total count not shown immediately after sharing?</h4>
It takes some time for the script to detect the sharing. So wait a few minutes than you see the total calculated clicks. Keep also in mind the caching time you defined in the admin panel.
So when you set the plugin to 5minutes caching time. You have to wait at least for 5minutes until the click count is shown.

<h4>Do i need a Mashshare or sharedcount account?</h4>
No you don´t. There is no account needed for up to 10.000 daily requests. For most websites this is suitable enough as Mashshare make use of exensive caching so 
the requests to sharedount are reduced to a little. If you want to have more often updated share counts you can register at sharecount for a free account and than are able
to use a free api key which increases your daily request limit up to 50.000 which should be fine for very large websites. Within the settings page of Mashshare Share buttons you find the sharedcount register link.

<h4>Does this plugin sends any personal user data to you or to Facebook, Twitter etc.?</h4>

No, there is no personal data send to Facebook, Twitter, Google and other services. There is also no data which goes to my hands that includes any IP or other data. 
The big advantage of using this Mashare Share buttons is the independance in comparision to other plugins which creates steady connections to Facebook and Co. 
So there is no IP based data send to the social networks or to sharedcount. 

<h4>How is sharedcount able to get the total number of shares?</h4>
Sharedcount is using public available API services of the social networks which deliver only the number of shares for a specific webpage. Sharedcount is not able to see who shared anything, only how often.

<h4>Do i have to do manual changes in Javascript or HTML Code?</h4>
There is no need for you to make any manual changes. The plugin does everything for you. But if you are an experienced web-developer you are free to use the php function mashsharer(); in your templates.

<h4>Is there a shortcode for pages and posts?</h4>
Use the shortcode [mashshare] to embed the Share Buttons in pages or posts.

<h4>Why do i get a blank page when i try to activate or uninstall the plugin</h4>
Most times this is a result of some php server limits. Check your logfiles to see which values must be increased.
I can also assist you with such problems, but keep in mind that they are mostly not caused by this plugin.

<h4>When i click on the share buttons, nothing happens and no popup window</h4>
Mostly that is because you are using any third party and outdated theme which is not coded very well by the author and is not using the Wordpress API for embeding external plugin script.
For Mashsharer make sure your website source contains the script /mashsharer/assets/mashsharer.js

So if you have no chance to update or change your theme do some hardcoding and put the following line into the head template of your theme file:
`<script type='text/javascript' src='http://yourwebsite.com/wp-content/plugins/mashsharer/assets/mashsharer.js?ver=1.1'></script>`

<h4>Why is Facebook only sharing the URL and not the title and description of my page?</h4>
Facebook does not longer supports custom titles, descriptions and images in the sharer.php but you can use open graph meta tags to show the desired custom formats. 
So if you theme does not suppport open graph meta tags (you see them in the html header as og: tags) use the Mashshare Add-On:
[Open Graph Social Optimization] (https://www.mashshare.net/downloads/mashshare-open-graph/)

After installation, check with the [Facebook Open graph debugger](https://developers.facebook.com/tools/debug/) how Facebook is seeing your site adn purge the Facebook Open Graph cache if necessary.


== Official Site ==
* https://www.mashshare.net

== Installation ==
1. Download the share button plugin "Mashshare" , unzip and place it in your wp-content/plugins/ folder. You can alternatively upload and install it via the WordPress plugin backend.
2. Activate the plugin through the 'Plugins' menu in WordPress.
3. Select Plugins->Mashshare

== Screenshots ==

1. Default share buttons (included)
2. Default share buttons with separate available responsive Add-On
3. Add-On Marketplace with free and premium Add-Ons
4. Default Sharebuttons with separate available responsive Add-On
5. Responsive Design + Social-Networks (separate Add-Ons)

changelog.txt
CHANGELOG MASHSHARE

= 2.1.2 =
Fix: Padding issue Whatsapp icon
Fix: Delete some deprecated code
Fix: More values for border radius to create circle like buttons
Fix: Gradiant style compatible in Firefox
New: New callback function colorpicker for settings
New: Performance benchmarks compaired with other plugins published in readme.txt

= 2.1.1 =
Fix: Some issues where js and css is not loaded on specific post_types and frontpage on third party plugins like event calender
New: WPMU WordPress Multisite compatible
New: Some js modifications for compatibility with Twitter Short-URLs Add-On
New: Gradiant share button style
New: CDN Compatibility - Embed icon fonts as base64 octled to prevent cross origin issues when website is loaded from CDN

= 2.1.0 =
Fix: Problem where sharecounts are zero

= 2.0.9 =
Fix: Allows Shortcodes in Subscribe forms!!!
Fix: Suppress header output warning when Debug mode is enabled
Fix: Prevent case where Social Networks are empty after updating from previous version
Fix: text-decoration: none !important for class mashsb-buttons a. This makes sure no third party class injects underline to the mashshare buttons
Fix: Minnor css fixes in backend
New: New class for sharedcount service 
New: Performance improvements: Use post_meta instead wpdb sql calls
New: New function for embeding Mashshare on all possible post_types

= 2.0.8 =
New: Better visual update message on plugins.php with parsing of readme.txt
New: Integration of Image uploader for Mashshare Open Graph Add-On
Tweak: Thinner descriptions and titles - CSS

= 2.0.7 =
New: Hide Sharecount until specific number of shares is reached
New: Updated readme.txt

= 2.0.6 =
New: Shortcodes enabled in all custom post_types
New: Create filter for share count 'mashsb_sharecount_filter'
New: Exclude Mashshare from specific posts and pages
Fix: Prevent getting shares with function getSharedcount() although share count is disabled

= 2.0.5 =
New: Check if Shortcodes is used, only than embed mashshare js and css files
New: Function to delete all database cached Mashshare objects
New: Define execution order of the plugin in comparision to other content plugins
New: Links in plugin meta row for "Getting Started" and "Add-Ons"
New: Tab for easier access to the Add-Ons

= 2.0.4 =
New: Shortcode alignment possible [mashshare align="left"] [mashshare align="right"]
Fix: Load js and css only on active share buttons pages and posts. 
Fix: Load Share buttons only on singular pages. Pages, Posts, Frontpage
Fix: Load Mashshare JS only on Mashshare admin settings page

= 2.0.3 =
New: Shortcodes: 
    Buttons with disabled sharecount: [mashshare shares="false" buttons="true"]
    Sharecount without buttons: [mashshare shares="true" buttons="false"]
    Buttons and Sharecount: [mashshare]
New: Enable debug mode
New: First redirect after installation goes to 'Social networks' tab
New: do_action('mashshare');
Fix: Change css path to MASHSB_PLUGIN_URL
Fix: Compatibility patch for old shortcode [mashshare]
Fix: Rename fontello fonts to mashsb-font.ttf to prevent multiple use of same icon font when page is using fontelle fonts for theme or other plugins
Fix: Count additional networks when Add-On is installed
Fix: remove some deprecated functions
Fix: Not default networks available

= 2.0.1/2 =
Fix: No share buttons visible
Fix: JS error. Check for undefined mashsb
Fix: Round Fake count factor

= 2.0.0 =
* New: Improved performance
* New: Option to disable share count completely  (no sql queries will be generated any longer)
* New: Shortcode option to disable share counts
* New: Check if curl is working on the server
* New: Option to disable share cache for testing purposes
* New: Use of sharp and crisp clear font icons instead png icons
* New: Button 'extra content' for content slider subcribe forms or any other content
* New: Use a link behind the Subscribe button instead the toggle slider
* New: Complete rewrite of css for easier modifications
* New: Improved extension system
* New: Improved backend, new Add-On page
* New: Multi language capable, *.po files
* New: Change color of share counts via setting
* New: Count up animation for share counts (Does not work for shortcodes and on blog pages)
* New: HTML5 Tag <aside> wrapped around to tell search engines that the share buttons are not part of the content
* New: Plus button moves to end of share buttons when activated and does not stay longer in place.
* New: Drag and drop sort order of services.
* New: Enable desired services with one click
* New: Choose which network should be visible all the time This ones will be large sized by default. Other ones are behind the plus sign
* New: Two different share button styles includes
* New: Choose border radius of the buttons from settings
* New: Keep settings when plugin is uninstalled - optional
* New: Custom CSS field
* Fix: Hashtags in post titles will be replaced. This fixes blank content issues for sharing on twitter

= 1.2.9 =
* Fix: void(0) instead return(0)
* Fix: Remove url_encode in twitter title
* New: Change Add-On Url to https://

= 1.2.8 =
* Fix: Show mashshare on frontpage in full blog layout
* New: Add Hashtag e.g. via @website

= 1.2.7 =
* Fix: Count issue for unshared posts

= 1.2.6 =
* Fix: Count G+, Linked, Pinterest, Stumble Upon when mashshare-networks installed.
* New: Add-On WPSSO

= 1.2.5 =
* New: Security Layer prevents zero share button sharecounts when sharedcount API is down 
* Fix: Change share button add_filter priority to 1000 to prevent errors with other plugins
* Fix: Delete non printable character which wraps the share buttons
* Fix: Remove duplicate quote in share button

= 1.2.4 =
* New: Option to round the shares e.g. 1.5k instead 1500

= 1.2.3 =
* Fix: linkedIn Sharebutton

= 1.2.2 =
* Share button Compatibility for WordPress 3.9.1

mashsharer-admin.php
<?php

/* Mashshare admin template
 * Author: René Hermenau
 * @since 1.0.0
 * @change 1.0.1
 */ 


/* Quit */
defined('ABSPATH') OR exit;

include_once 'class.mashsharer.php';
include_once 'class.debug.php';

/* Define Constants */
global $wp_version;

add_action( 'admin_menu', 'mashsharer_admin_menu' );

function mashsharer_admin_menu() {
		mashsharer_load_menu();
}

function mashsharer_load_menu() {
	$current_theme = wp_get_theme();
	$pluginpath = MASHSHARER_PLUGIN_INSTALL_FILE;
	$active_plugin = get_option('active_plugins');
	
	$plugin_key = array_keys($active_plugin,"$pluginpath");
	$active_plugin_key = $plugin_key[0];
	
	add_submenu_page('options-general.php', __('Mashshare'), __('Mashshare'), 'manage_options', 'mashsharer-config', 'mashsharer_conf');
	
}

	

function mashsharer_admin_init() {
    global $wp_version;
    // all admin functions are disabled in old versions
    if (version_compare( $wp_version, '3.0', '<' ) ) {
        
        function mashsharer_version_warning() {
			echo "<div id='mashsharer-warning' class='updated fade'><p><strong>".sprintf(__('Mashsharer %s requires WordPress 3.0 or higher.'), MASHSHARER_VERSION_NUM) ."</strong> ".sprintf(__('Please <a href="%s">upgrade WordPress</a> to a current version.'), 'http://codex.wordpress.org/Upgrading_WordPress'). "</p></div>";
		}
		add_action('admin_notices', 'mashsharer_version_warning');
		return;    
    }
}
add_action('admin_init', 'mashsharer_admin_init');


function mashsharer_nonce_field($action = -1) { return wp_nonce_field($action); }
$mashsharer_nonce = 'mashsharer-update-key';


/**
	* Rückgabe der Optionen
	*
	* @since   1.0.0.
	* @change  
	*
	* @return  array  $diff  Array mit Werten
	*/

	function get_options()
	{
		return wp_parse_args(
                        
			array(
				'apikey'	 	=> get_option('mashsharer_apikey'),
				'cache'		 	=> get_option('mashsharer_cache_expire'),
                'position'		=> get_option('mashsharer_position'),
                'pages'                 => get_option('mashsharer_pages'),
                'posts'                 => get_option('mashsharer_posts'),
                'frontpage'             => get_option('mashsharer_frontpage'),
				'round'					=> get_option('mashsharer_round'),
				'hashtag'				=> get_option('mashsharer_hashtag'),
           			)
		);
	}
        
        /**
	* Position dropdown
	*
	* @since   1.0.0
	* @change  
	*
	* @return  array    Key => value array
	*/

	function position_select()
	{
		return array(
			'before'   => 'Top',
			'after'   => 'Bottom',
			'both'   => 'Top and Bottom',
            'manual'   => 'Manual'
		);
	}

/**
	* Verfügbare Cache-Methoden
	*
	* @since  2.0.0
	* @change 2.1.3
	*
	* @param  array  $methods  Array mit verfügbaren Arten
	*/

        function method_select()
	{
		/* Defaults */
		$methods = array(
			'300' => 'in 5 minutes',
			'600' => 'in 10 minutes',
			'1800' => 'in 30 minutes',
            '3600' => 'in 1 hour',
			'21600' => 'in 6 hours',
            '43200' => 'in 12 hours',
            '86400' => 'in 24 hours'
		);

		return $methods;
	}
        
        



/* Update and save configuration */
function mashsharer_conf() {
	global $mashsharer_nonce;
        
	if ( isset($_POST['submit']) ) {
		if ( function_exists('current_user_can') && !current_user_can('manage_options') )
			die(__('Cheatin&#8217; uh?'));

		check_admin_referer( $mashsharer_nonce );

		if ( isset( $_POST['mashsharer_cache'] ) )
			update_option( 'mashsharer_cache_expire', $_POST['mashsharer_cache'] );
		
		if ( isset( $_POST['mashsharer_apikey'] ) )
			update_option( 'mashsharer_apikey', $_POST['mashsharer_apikey'] );
		
		if (isset( $_POST['mashsharer_posts']) == 1){
			update_option( 'mashsharer_posts', 1 ); 
                } else{ 
                        update_option( 'mashsharer_posts', 0 );
                }
                
                if ( $_POST['mashsharer_pages'] == 1){
			update_option( 'mashsharer_pages', 1 );
                } else {
                        update_option( 'mashsharer_pages', 0 ); 
                }

                 if ( isset( $_POST['mashsharer_position'] ) )
			update_option( 'mashsharer_position', $_POST['mashsharer_position'] );
                 
                 if ( $_POST['mashsharer_frontpage'] == 1){
			update_option( 'mashsharer_frontpage', 1 );
                } else {
                        update_option( 'mashsharer_frontpage', 0 );
                }
				
                 if ( $_POST['mashsharer_round'] == 1){
			update_option( 'mashsharer_round', 1 );
                } else {
                        update_option( 'mashsharer_round', 0 );
                }
				
				if ( isset( $_POST['mashsharer_hashtag'] ) )
			update_option( 'mashsharer_hashtag', $_POST['mashsharer_hashtag'] );
                

	} 

?>
<?php if ( !empty($_POST['submit'] ) ) : ?>
<div id="message" class="updated fade"><p><strong><?php _e('Options saved.') ?></strong></p></div>
<?php endif; ?>

	<div class="wrap rm_wrap">
            <div>
                <!-- TO DO <img src="<?php echo plugin_dir_url( __FILE__ );?>./images/logo.png">-->
            </div>
		<h2><?php _e('Mashsharer'); ?></h2>
		<div class="rm_opts">
			<form action="" method="post" id="mashsharer-conf" style="margin: auto;">
                            <?php $options = get_options() ?>
                            <table class="form-table"> 
                                <tr>
                                    <th scope="row">Cache expire</th>
                                        <td>
                                            <label for="mashsharer_cache_method">
                                                    <select name="mashsharer_cache" id="mashsharer_cache_method">
                                                        <?php foreach (method_select() as $k => $v) { ?>
                                                            <option value="<?php echo esc_attr($k) ?>" <?php selected($options['cache'], $k); ?>><?php echo esc_html($v) ?></option>
                                                        <?php } ?>
                                                    </select>
                                                </label>
                                        </td>
                                        <td>
                                         The amount of shares are updated after time of "cache expire". Notice that Sharedcount.com uses his own cache (30 - 60min) so it does not update immediately when expire time is very low, e.g. 5 minutes.
                                        </td>
                                    </tr>
                                    <tr valign="top">
                                        <th scope="row">API KEY - IMPORTANT!</th> 
                                        <td>
                                            <fieldset>
                                                    <label for="mashsharer-apikey">
                                                        <input type="text" name="mashsharer_apikey" id="mashsharer-apikey" value="<?php echo $options['apikey'];?>">
                                                    </label>
                                                </fieldset>
                                        </td>
                                        <td>
                                          Get it FREE at <a href="https://admin.sharedcount.com/admin/signup.php" target="_blank">SharedCount.com</a> for 50.000 free daily requests.
										  Its essential for accurate function of this plugin.
                                        </td>
                                    </tr>
                                    <tr valign="top">
                                    <th scope="row">Position</th>
                                        <td valign="top">
                                            <label for="mashsharer_position">
                                                    <select name="mashsharer_position" id="mashsharer_position">
                                                        <?php foreach (position_select() as $k => $v) { ?>
                                                            <option value="<?php echo esc_attr($k) ?>" <?php selected($options['position'], $k); ?>><?php echo esc_html($v) ?></option>
                                                        <?php } ?>
                                                    </select>
                                                </label>
                                            
                                        </td>
                                        <td valign="top">
                                             Choose where you would like the social icons to appear, before or after the main content. If set to Manual, you can use this code to place your Social links anywhere you like in your templates files:
                                             &nbsp; <strong>&lt;?php mashsharer(); ?&gt;</strong> or use the shortcode: [mashshare] in your posts. <strong>Note: You have to use permanent urls</strong>
                                        </td>
                                        <tr valign="top">
						<th scope="row">
							<?php _e('Post Types', 'mashsharer') ?>
						</th>
						<td>
							<fieldset>
								<label for="mashsharer_posts">
									<input type="checkbox" name="mashsharer_posts" id="mashsharer_posts" value="1" <?php checked('1', $options['posts']); ?> />
									
								</label>
                                                                Posts
								<br />

								<label for="mashsharer_pages">
									<input type="checkbox" name="mashsharer_pages" id="mashsharer_pages"  value="1" <?php checked('1', $options['pages']); ?> />
									
								</label>
                                                                Pages
							</fieldset>
						</td>
                                                <td>
                                                    <?php _e('Select where you want the share buttons appear', 'mashsharer') ?>
                                                </td>
					</tr>
                                        <tr valign="top">
						<th scope="row">
							<?php _e('Frontpage', 'mashsharer') ?>
						</th>
						<td>
							<fieldset>
								<label for="mashsharer_frontpage">
									<input type="checkbox" name="mashsharer_frontpage" id="mashsharer_frontpage" value="1" <?php checked('1', $options['frontpage']); ?> />
								</label>
                                                         </fieldset>
						</td>
                                                <td>
                                                    <?php _e('Select when the buttons should appear on the frontpage', 'mashsharer') ?>
                                                </td>
					</tr>
					<tr valign="top">
						<th scope="row">
							<?php _e('Round Share counts ', 'mashsharer') ?>
						</th>
						<td>
							<fieldset>
								<label for="mashsharer_round">
									<input type="checkbox" name="mashsharer_round" id="mashsharer_round" value="1" <?php checked('1', $options['round']); ?> />
									
								</label>
                                                                Let´s round it!
								
                                                                
							</fieldset>
						</td>
                                                <td>
                                                    <?php _e('E.g.: Shares greater than 1000 are shown like 1k', 'mashsharer') ?>
                                                </td>
					</tr>
					<tr valign="top">
                                        <th scope="row">Twitter Hashtag</th> 
                                        <td>
                                            <fieldset>
                                                    <label for="mashsharer-hashtag">
                                                        <input type="text" name="mashsharer_hashtag" id="mashsharer-mashtag" value="<?php echo $options['hashtag'];?>">
                                                    </label>
                                                </fieldset>
                                        </td>
                                        <td>
                                          Optional: Use name of your website or any other hashtah, e.g. 'Mashshare' results in via @Mashshare
                                        </td>
                                    </tr>
                                        <tr valign="top">
						<th scope="row">
							<?php _e('Support', 'mashsharer') ?>
						</th>
						<td>
							
						</td>
                                                <td>
                                                    <?php _e('Do you have any issues? Write me and i try my best to fix it: <a href="mailto:rene@digitalsday.com">rene@digitalsday.com</a>.<br> Please also rate this plugin at <a href="http://wordpress.org/support/view/plugin-reviews/mashsharer?filter=5" target="_blank">wordpress.org</a>. That motivates me to improve the plugin.', 'mashsharer') ?>
                                                </td>
					</tr>
                                    </tr>
                                </table>    
                            
				
                          
                           
                            <div style="float:clear;">
                                
                                <p class="submit"><input type="submit" name="submit" value="<?php _e('Save settings &raquo;'); ?>" /></p>
                            </div>
                         <?php mashsharer_nonce_field($mashsharer_nonce); ?>
			</form>

		</div>
	</div>
<div style="clear:both;">
    <?php mash_add_ons_page(); ?>
<div style="clear:both;">

</div>
<?php

}



?>